lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1794,6 +1836,84 @@ def cmd_sessions(*args)
     true
   end
 
+  def get_matching_sessions(search_term)
+    matching_sessions = {}
+    terms = search_term.split
+    terms.each do |term|
+      matches = filter_sessions_by_search(term)
+      matches.each do |session_id, session|
+        matching_sessions.store(session_id, session)
+      end
+    end
+    matching_sessions
+  end
+
+  def filter_sessions_by_search(search_term)
+    matching_sessions = {}
+    
+    framework.sessions.each do |session_id, session|
+      case search_term.split(":")[0]
+      when "last_checkin"
+        if session.respond_to?(:last_checkin) && session.last_checkin && evaluate_search_criteria(session, search_term)
+          matching_sessions.store(session_id, session)
+        end
+      when "session_type"
+        matching_sessions.store(session_id, session) if evaluate_search_criteria(session, search_term)
+      when "session_id"
+        matching_sessions.store(session_id, session) if evaluate_search_criteria(session, search_term)
+      end
+    end
+    matching_sessions
+  end
+
+  def evaluate_search_criteria(session, search_term)
+    parts = search_term.split(":")
```

Comment:
```suggestion
    field, operator, value = search_term.split(":")
```
example of how this works:
```
[6] pry()> a = "x:y:z"
=> "x:y:z"
[7] pry()> b = "q:r"
=> "q:r"
[8] pry()> field, operator, value = a.split(":")
=> ["x", "y", "z"]
[9] pry()> field
=> "x"
[10] pry()> operator
=> "y"
[11] pry()> value
=> "z"
[12] pry()> field1, operator1, value1 = b.split(":")
=> ["q", "r"]
[14] pry()> field1
=> "q"
[15] pry()> operator1
=> "r"
[16] pry()> value1
=> nil
[17] pry()>
```